packages/embark-mocha-tests/src/lib/index.js

```diff
@@ -0,0 +1,151 @@
+    this.embark = embark;
+
+    this.files = [];
+
+    const { events } = embark;
+    events.request('tests:runner:register',
+      'JavaScript (Mocha)',
+      this.match.bind(this),
+      this.addFile.bind(this),
+      this.run.bind(this)
+    );
+  }
+
+  addFile(path) {
+    if (!this.match(path)) {
+      throw new Error(`invalid JavaScript test path: ${path}`);
+    }
+
+    this.files.push(path);
+  }
+
+  match(path) {
+    return JAVASCRIPT_TEST_MATCH.test(path);
+  }
+
+  run(options, cb) {
+    const {events} = this.embark;
+
+    const Module = require("module");
+    const originalRequire = require("module").prototype.require;
+
+    let accounts = [];
+    let compiledContracts = {};
+    let web3;
+
+    const config = (cfg, acctCb) => {
+      global.before((done) => {
+        async.waterfall([
+          (next) => {
+            events.request("contracts:build", cfg, compiledContracts, next);
+          },
+          (contractsList, contractDeps, next) => {
+            events.request("deployment:contracts:deploy", contractsList, contractDeps, next);
+          },
+          (_foo, next) => {
+            events.request("contracts:list", next);
+          },
+          (contracts, next) => {
+            for(const contract of contracts) {
+              const instance = compiledContracts[contract.className];
+              const contractObj = new web3.eth.Contract(instance.abiDefinition, contract.deployedAddress);
+              Object.setPrototypeOf(compiledContracts[contract.className], contractObj);
+            }
+
+            next();
+          }
+        ], (_err) => {
+            if (acctCb) {
+              acctCb(null, accounts);
+            }
+
+            done();
+        });
+      });
+    };
+
+    async.waterfall([
+      (next) => { // request provider
+        events.request("blockchain:client:provider", "ethereum", next);
+      },
+      (bcProvider, next) => { // set provider
+        web3 = new Web3(bcProvider);
+        next();
+      },
+      (next) => { // get accounts
+        web3.eth.getAccounts((err, accts) => {
+          if (err !== null) {
+            return next(err);
+          }
+
+          accounts = accts;
+          next();
+        });
+      },
+      (next) => { // reset contracts as we might have state leakage from other plugins
+        events.request("contracts:reset", next);
+      },
+      (next) => { // get contract files
+        events.request("config:contractsFiles", next);
+      },
+      (cf, next) => { // compile contracts
+        events.request("compiler:contracts:compile", cf, next);
+      },
+      (cc, next) => { // override require
+        compiledContracts = cc;
+
+        Module.prototype.require = function(req) {
+          const prefix = "Embark/contracts/";
+          if (!req.startsWith(prefix)) {
+            return originalRequire.apply(this, arguments);
+          }
+
+          const contractClass = req.replace(prefix, "");
```

Comment:
`require("Embark/contracts/foo")` should be not found in the contracts array if the contract doesn't exist in the DApp obviously. But it might be a good idea for an error to be thrown so that the test fails.